RecoEgamma/EgammaPhotonProducers/src/GEDPhotonProducer.cc

```diff
@@ -946,20 +946,25 @@ void GEDPhotonProducer::fillPhotonCollection(edm::Event& evt,
       if (candidateP4type_ == "fromEcalEnergy") {
         newCandidate.setP4(newCandidate.p4(reco::Photon::ecal_photons));
         newCandidate.setCandidateP4type(reco::Photon::ecal_photons);
+        newCandidate.setMass(0);
       } else if (candidateP4type_ == "fromRegression1") {
         newCandidate.setP4(newCandidate.p4(reco::Photon::regression1));
         newCandidate.setCandidateP4type(reco::Photon::regression1);
+        newCandidate.setMass(0);
       } else if (candidateP4type_ == "fromRegression2") {
         newCandidate.setP4(newCandidate.p4(reco::Photon::regression2));
         newCandidate.setCandidateP4type(reco::Photon::regression2);
+        newCandidate.setMass(0);
       } else if (candidateP4type_ == "fromRefinedSCRegression") {
         newCandidate.setP4(newCandidate.p4(reco::Photon::regression2));
         newCandidate.setCandidateP4type(reco::Photon::regression2);
+        newCandidate.setMass(0);
       }
     } else {
       math::XYZVector gamma_momentum = direction.unit() * scRef->energy();
       math::XYZTLorentzVectorD p4(gamma_momentum.x(), gamma_momentum.y(), gamma_momentum.z(), scRef->energy());
       newCandidate.setP4(p4);
+      newCandidate.setMass(0);
```

Comment:
use PolarLorentzVector. This is the internal representation of reco candidate p4

Reply:
Hello Slava,

The setMass function does this transformation.
https://cmssdt.cern.ch/lxr/source/DataFormats/Candidate/interface/ParticleState.h#0159

Do you mean to use,
      math::PtEtaPhiMLorentzVector p4(gamma_momentum.Pt(), gamma_momentum.Eta(), gamma_momentum.Phi(), 0.0);
I could not find Pt function in math::XYZVector.

Please let me know your opinion.
Thanks,
Reza

Reply:
yes, I meant to replace the XYZT p4 construction and setP4;setMass with just a construction of Polar p4 followed by setP4

Reply:
Hi Slava,

As I mentioned, math::XYZVector does not have the Pt function. I need to recalculate it and I afraid to introduce bug there. What is wrong with using setMass function?
 It does this transformation automatically using setCartesian function,
 https://cmssdt.cern.ch/lxr/source/DataFormats/Candidate/interface/ParticleState.h#0159

Thanks,
Reza